Docker.DotNet/Microsoft.Net.Http.Client/ManagedHandler.cs

```diff
@@ -16,13 +16,23 @@ namespace Microsoft.Net.Http.Client
     public class ManagedHandler : HttpMessageHandler
     {
         public delegate Task<Stream> StreamOpener(string host, int port, CancellationToken cancellationToken);
+        public delegate Task<Socket> SocketOpener(string host, int port, CancellationToken cancellationToken);
 
-        public ManagedHandler(StreamOpener opener = null)
+        public ManagedHandler(StreamOpener opener)
         {
-            _opener = opener;
-            if (_opener == null)
+            _streamOpener = opener;
+            if (_streamOpener == null)
             {
-                _opener = TCPStreamOpener;
+                _socketOpener = TCPSocketOpener;
+            }
+        }
+
+        public ManagedHandler(SocketOpener opener = null)
```

Comment:
Seems odd to me. The compiler shouldn't know how to decode this style of default constructor.

IE:
```C#
new ManagedHandler() -> (SocketOpener default)
new ManagedHandler(null) -> err
new ManagedHandler((StreamOpener)null) -> (StreamOpener)
new ManagedHandler((SocketOpener)null) -> (SocketOpener)
```

Doesnt it make more sense to just say nulls aren't allowed and create an explicit default constructor that sets `TCPSocketOpener` to be more clear?